src/main/java/edu/harvard/iq/dataverse/search/SolrIndexServiceBean.java

```diff
@@ -547,7 +546,7 @@ public JsonObjectBuilder deleteAllFromSolrAndResetIndexTimes() throws SolrServer
             Timestamp permissionModificationTime = dvObject.getPermissionModificationTime();
             Timestamp permissionIndexTime = dvObject.getPermissionIndexTime();
             if (permissionIndexTime == null) {
-                if (dvObject.getId() != rootDvId) {
+                if (dvObject.getId() != rootDvId && !dvObject.isInstanceofDataFile()) {
```

Comment:
As a possible performance optimization, since we're skipping data files anyway, perhaps instead of the `dvObjectService.findAll()` above we could get just a list of dataverses and datasets. I don't know if it would help with performance or not, or if performance is even a problem.

Reply:
talked about this and decided to see how performance is in production before making a change